src/main/java/org/springframework/data/couchbase/core/query/Query.java

```diff
@@ -254,6 +254,16 @@ public String toN1qlString(ReactiveCouchbaseTemplate template, Class domainClass
 		return statement.toString();
 	}
 
+	public String toN1qlRemoveString(ReactiveCouchbaseTemplate template, Class domainClass) {
+		StringBasedN1qlQueryParser.N1qlSpelValues n1ql = getN1qlSpelValues(template, domainClass, false);
+		final StringBuilder statement = new StringBuilder();
+		appendString(statement, n1ql.delete); // delete ...
+		appendWhereString(statement, n1ql.filter); // typeKey = typeValue
+		appendWhere(statement, null); // criteria on this Query
```

Comment:
If the second arg to appendWhere() is  new int[] { 0 }   (as in toN1qlSelectString), the variables in the criteria will be parameterized for efficiency.

Reply:
Initially I had the same idea but in that case positional parameters in query will not be replaced with actual values. Also, it seems in current code `findByQuery(clazz).matching(query)` method is not behaving as expected (added new test case in `CouchbaseTemplateQueryIntegrationTests` in this PR which is failing).

Reply:
I don't think we should submit a pr with a failing test - maybe submit it in a different PR so we can work on a fix alongside it?

Reply:
Yes, I can create failing test in separate PR. The question is how to handle comment from @mikereiche where he proposes to use `new int[] {0}` which does not work at the moment?

Reply:
New bug created https://jira.spring.io/browse/DATACOUCH-619. 
Resolving initial comment without applying `new int[] {0}` as it is not working as expected.